exp/controllers/rosamachinepool_controller.go

```diff
@@ -250,22 +233,60 @@ func (r *ROSAMachinePoolReconciler) reconcileNormal(ctx context.Context,
 	}
 
 	npBuilder.AWSNodePool(cmv1.NewAWSNodePool().InstanceType(rosaMachinePool.Spec.InstanceType))
-	if rosaMachinePool.Spec.Version != "" {
-		npBuilder.Version(cmv1.NewVersion().ID(ocm.CreateVersionID(rosaMachinePool.Spec.Version, ocm.DefaultChannelGroup)))
-	}
 
 	nodePoolSpec, err := npBuilder.Build()
```

Comment:
Let's do the optional version addition above this and only call `.Build()` once

Reply:
the version field can't be part of the nodepool Update call. So its only set if we are creating not updating.

Reply:
I see, I missed the new `Update()` flow. As noted in Slack it seems like OCM would be expecting us to do a diff and send updates only as needed, let's make those changes.

If you can determine a more clear codepath here it would be good. Right now we have a builder, call a `.Build()` on it, then re-use the builder and re-build. The control flow of "create half a builder, get the current object, maybe update it, finish the builder, call create" is difficult to follow. It is not entirely clear that this is safe, expected, or normal. Other builder patterns will reclaim memory or otherwise change internally when `.Build()` is called so even aside from the control flow, it's surprising to see it.